app/models/manageiq/providers/redhat/infra_manager.rb

```diff
@@ -1,18 +1,18 @@
 class ManageIQ::Providers::Redhat::InfraManager < ManageIQ::Providers::InfraManager
-  require_nested :EventCatcher
-  require_nested :EventParser
-  require_nested :RefreshWorker
-  require_nested :RefreshParser
-  require_nested :MetricsCapture
-  require_nested :MetricsCollectorWorker
-  require_nested :Host
-  require_nested :Provision
-  require_nested :ProvisionViaIso
-  require_nested :ProvisionViaPxe
-  require_nested :ProvisionWorkflow
-  require_nested :Refresher
-  require_nested :Template
-  require_nested :Vm
+  require_nested  :EventParser
+  require_nested  :RefreshWorker
+  require_nested  :RefreshParser
+  require_nested  :MetricsCapture
+  require_nested  :MetricsCollectorWorker
+  require_nested  :Host
+  require_nested  :Provision
+  require_nested  :ProvisionViaIso
+  require_nested  :ProvisionViaPxe
+  require_nested  :ProvisionWorkflow
+  require_nested  :Refresher
+  require_nested  :Template
+  require_nested  :Vm
+  include_concern :ApiIntegration
```

Comment:
I am not sure what to do with this, i mean I think it should definitely move to a separate module (actually I think it should be built differently and move to a different class all together) and I could not really work on this when it was all mixed. So I should have moved it first, open a PR on that and only then do the rest. But I did not. If its crucial I will do it now. Is it? @durandom